clang/lib/CIR/CodeGen/CIRRecordLayoutBuilder.cpp

```diff
@@ -630,7 +630,7 @@ CIRGenTypes::computeRecordLayout(const RecordDecl *D,
 
   auto RL = std::make_unique<CIRGenRecordLayout>(
       Ty ? *Ty : mlir::cir::StructType{},
-      BaseTy ? *BaseTy : mlir::cir::StructType{},
+      BaseTy ? BaseTy : mlir::cir::StructType{},
```

Comment:
> we were rewriting the type being processed (due tp misused pointers).

Is this the "misused pointer" you've mentioned? These seem like non-functional changes.